packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -1091,22 +1093,24 @@ export function filterSuperfluousBuilds(
 			}
 
 			if (innerMap.size === 0) {
-				builds.delete(revision);
+				refreshers.delete(revision);
 			}
 		} else {
-			// if this revision does not exist in the relevant removed roots, delete it from the builds
-			builds.delete(revision);
+			// if this revision does not exist in the relevant removed roots, delete it from the refreshers
+			refreshers.delete(revision);
 		}
 	}
 
-	const { fieldChanges, maxId, revisions, constraintViolationCount, destroys } = change.change;
+	const { fieldChanges, maxId, revisions, constraintViolationCount, builds, destroys } =
+		change.change;
 	return makeModularChangeset(
 		fieldChanges,
 		maxId,
 		revisions,
 		constraintViolationCount,
-		builds.size > 0 ? builds : undefined,
+		builds,
 		destroys,
+		refreshers.size > 0 ? refreshers : undefined,
```

Comment:
The `refreshers.size > 0` check is already performed by `makeModularChangeset` so you don't need it here.